src/backend/api/Fusion.Resources.Api/Startup.cs

```diff
@@ -0,0 +1,106 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Fusion.Resources.Api
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                .AddJwtBearer(options =>
+                {
+                    options.Audience = Configuration["AzureAd:ClientId"];
+                    options.Authority = "https://login.microsoftonline.com/3aa4a235-b6e2-48d5-9195-7fcf05b459b0";
+                    options.SaveToken = true;
+                });
+
+
+            services.AddHttpContextAccessor();
+            services.AddSwagger(Configuration);
+
+            // Configure fusion integration
+            services.AddFusionIntegration(options =>
+            {
+                options.AddFusionAuthorization();
+
+                options.UseDefaultEndpointResolver("ci");
+                options.UseDefaultTokenProvider(opts =>
+                {
+                    opts.ClientId = Configuration["AzureAd:ClientId"];
+                    opts.ClientSecret = Configuration["AzureAd:ClientSecret"];
+                });
+
+                options.ApplicationMode = true;
+            });
+
+            services.AddControllers();
+
+            services.AddHealthChecks()
+                .AddCheck("liveness", () => HealthCheckResult.Healthy());
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            app.UseCors(opts => opts
+                .AllowAnyOrigin()                
+                .AllowAnyMethod()
+                .AllowAnyHeader());
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseResourcesApiSwagger(Configuration);
+
+            app.UseAuthentication();
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+
+            #region Health probes
+
+            app.UseHealthChecks("/_health/liveness", new HealthCheckOptions
+            {
+                Predicate = r => r.Name.Contains("liveness")
+            });
+            app.UseHealthChecks("/_health/ready", new HealthCheckOptions
+            {
+                Predicate = r => r.Name.Contains("liveness")
```

Comment:
Will `r.Name` contain `"liveness"` in this case? Shouldn't it be `"ready"`?

Reply:
that predicate defined which tests, defined in the configure services section, to execute for that endpoint.